test/clj_kondo/main_test.clj

```diff
@@ -740,10 +740,15 @@ foo/foo ;; this does use the private var
   (assert-submaps
    '({:file "<stdin>",
       :row 1,
-      :col 1,
+      :col 9,
+      :level :warning,
+      :message "Missing else branch."}
+     {:file "<stdin>",
+      :row 1,
+      :col 35,
       :level :warning,
       :message "Missing else branch."})
-   (lint! "(if-let [x 1] true)"))
+   (lint! "#?(:clj (if-let [x 1] true) :cljs (if-let [x 1] true))" "--lang" "cljc"))
```

Comment:
Actually I think the reader conditionals aren't needed here: code outside reader conditionals counts for all hosts, so if one of them would generate a an error, you would still see it.

Reply:
E.g.:

```
$ clj-kondo --lint - --lang cljc <<< 'future'
<stdin>:1:1: error: unresolved symbol future
```

We are getting this error because of CLJS, even though it was valid in CLJ.

Reply:
I need not "one of them" but "both of them" to generate an error, otherwise this test won't spot regressions where the cljs case is omitted. Hopefully I have not misunderstood.

Reply:
This lints the code unconditionally as both CLJ and CLJS, because it is not in a conditional:

```
clj-kondo --lint - --lang cljc <<< 'future'
```

Reply:
When I change the test to:
```
   (lint! "(if-let [x 1] true)" "--lang" "cljc")
```
and remove the second asserted submap, the test does pass, but if I comment out the `[cljs.core if-let]` from the cases in `lint-specific-calls!` the test _still_ passes, and it shouldn't, because I want to test against such a regression.

Reply:
@tomdl89 I understand that, but `(if-let [x 1] true)` is exactly the same as `#?(:clj (if-let [x 1] true) :cljs (if-let [x 1] true))`. It might make sense to pull the test apart for each lang and not use cljc then.

Just use a variable and doseq:

```
(doseq [lang ["clj" "cljs"]]
  (is ...))
```

Reply:
Actually, it's not exactly the same, because the former outputs one result, and the latter outputs two results. Indeed, if I comment out the cljs case in `lint-specific-calls!` with the reader-conditional approach I get this error:
```
Expected 2 results, but got: 1
```
so it works well at capturing the regression I'm targeting. I'm happy to use the doseq approach if you feel it's more readable? Sorry to be contrary - just want to make sure I implement a solid test for this 🙂

Reply:
Let's do the `doseq` approach since it feels more robust.